aten/src/ATen/native/mps/kernels/EmbeddingBag.metal

```diff
@@ -22,55 +22,65 @@ struct ReductionOpInit<EmbeddingBagMode::MAX, T> {
 
 template <EmbeddingBagMode M, typename T>
 struct ReductionOp {
+  inline opmath_t<T> operator()(opmath_t<T> weight_val, opmath_t<T> out_val) {
+    return weight_val + out_val;
+  }
+};
+
+template <typename T>
+struct ReductionOp<EmbeddingBagMode::MAX, T> {
+  inline opmath_t<T> operator()(opmath_t<T> weight_val, opmath_t<T> out_val) {
+    return max(weight_val, out_val);
```

Comment:
Q: Which max are you using there? one from `metal::` or one from `c10::metal::`? 
I don't know if embedding bug supposed to carry about NaN, but if it is, make sure to use `c10::metal::` wrapper, as regular one will not be able to handle NaN correctly

Reply:
Good catch. Looks like the only way to make it match the CPU impl is to use `metal::max` and also initialize `out_val` with nan. I'll make those changes and add some nans to the test

Reply:
Ok I've updated it to have the same nan behavior as the CPU impl. It doesn't use `max` at all, and instead uses comparison. Let me know what you think

Reply:
By the way, I don't know which one is faster? nans or comparison